### DISCUSSION TITLE
How to Visit Unreleased or Inaccessible Maps

### DISCUSSION THREAD

**Discussion by NealSpellman** [DarkflameUniverse/DarkflameServer#1554] [General]
Hello Explorers!

Today we'll be talking about how to visit maps that are not normally accessible or were brought in from older clients. One great example of an inaccessible map that still is in the client is Frostburgh; this map was made inaccessible in `cdclient.fdb`, with its' missions also stripped from the client. There are two ways to get your client to load maps like Frostburgh;

1. Reuse an existing zone entries' file path (The Simple Way)
2. Manually fix the game's database (The Correct Way)

Before starting, we recommend you make a copy of `cdclient.fdb` and convert it to SQLite using [this tool.](https://github.com/IAmMajo/FDB-Converter) There are many open-source SQLite editors, such as [sqlitebrowser.](https://github.com/sqlitebrowser/sqlitebrowser)

**The Simple Way**
The easiest way to add a world is the following;

1. Open `cdclient.sqlite` and navigate to the Zone table.
2. Find an entry that is unused but kept its' file path, such as Lightning Gauntlet.
3. Create a new folder in your client and on your server matching that unused zone's file path. (In this example, you'd create a folder in `/maps/` called `lightninggauntlet`)
4. Once you've created this folder, check the name of the luz file mentioned in the unused zone's file path. Note this down.
5. Copy over the desired map files (the luz, lvls, raw, etc of the world you want to visit) into the `/lightninggauntlet/` folder.
6. Rename the luz file in this folder on both the client and the server to match the unused zone's one we looked at in step 5. (In this example, you'd rename the luz to `nd_nj_lightning.luz`)
7. Launch the client and server, then `/testmap` to the zone ID matching the zone you chose to take the filepath from. (If you used `lightninggauntlet`, the zone ID is 201)

You should now be able to load into the map, though it may not load if it is missing other necessary files in the client. Double check all client logs and troubleshoot any errors impacting loading. If you'd like to play on these maps with friends on your server, you'll need to make sure all players have done the same changes to their client.

**The Correct Way**
If you'd like to add a world in the correct way, we suggest utilizing tools to edit `cdclient.sqlite` and, if the world is still present in the Zone table in some fashion, editing the entry to match your desired client files. If it isn't, you can utilize one of the many test/developer entries in the Zone table and edit it to match your imported map.

Once you've made the changes in your `cdclient.sqlite`, convert it back to FDB and put it into your client. You will need to make similar changes in your server's database as well. Each player will need your updated `cdclient.fdb` as well.

For associated content that was stripped out such as Frostburgh's missions and other older content, you can splice in its' data if you own a client that includes it. This process requires advanced knowledge of the LEGO Universe client.

**Disclaimer**
Given that these maps are not accessible by default, we give no guarantee on content working or offer any support for them. Many maps that exist outside of the accessible bunch are broken in ways that would require client patches and server changes.

We hope this helps and that players enjoy visiting any maps they have.

**Comment by Jettford** [comment]
> Before starting, we recommend you make a copy of cdclient.fdb and convert it to SQLite using [this tool.](https://github.com/IAmMajo/FDB-Converter) There are many open-source SQLite editors, such as [sqlitebrowser.](https://github.com/sqlitebrowser/sqlitebrowser)

It would be better to link the site than the Github for the site: https://fdb.lu-dev.net/
